geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-api/src/main/scala/org/locationtech/geomesa/fs/storage/api/PartitionScheme.scala

```diff
@@ -49,6 +60,34 @@ trait PartitionScheme {
     * @return list of simplified filters and partitions
     */
   def getSimplifiedFilters(filter: Filter, partition: Option[String] = None): Option[Seq[SimplifiedFilter]] = None
+
+  /**
+   * Get partitions that intersect the given filter
+   *
+   * If the filter does not constrain partitions at all, then an empty option will be returned. If
+   * the filter excludes all potential partitions, then an empty list will be returned
+   *
+   * @param filter filter
+   * @return list of intersecting filters
+   */
+  def getIntersectingPartitions(filter: Filter): Option[Seq[String]] = {
+    // default impl to prevent API breakage in partition scheme impls
+    // TODO remove default impl in next major release
+    throw new NotImplementedError()
+  }
+
+  /**
+   * Get a filter that will cover a partitions, i.e. the filter will return all features
+   * in the given partition and none from other partitions
+   *
+   * @param partition partition to cover
+   * @return filter
+   */
+  def getCoveringFilter(partition: String): Filter = {
+    // default impl to prevent API breakage in partition scheme impls
+    // TODO remove default impl in next major release
```

Comment:
Can you add a ticket for this API clean and tag for 4.0.0?

Reply:
added https://geomesa.atlassian.net/browse/GEOMESA-2966